accepted/2021/workloads/workload-installation.md

```diff
@@ -0,0 +1,337 @@
+
+- For in-place updates to the .NET SDK, we should also update the workloads when we update the SDK
+  - For SDK installations managed by Visual Studio, in-place updates can include updating to a new SDK feature band.  In that case the workload installation is managed by Visual Studio, so the workloads will be updated as we want with the plans we have today
+- When installing a new version of the .NET SDK side-by-side with an existing version, we should default to installing the same workloads that were installed for the previous version.  The developer should be able to override this, for example by unchecking the corresponding checkboxes in an installer UI
+
+Given that for some installers we won't be able to automatically, we should make it easy to install the same workloads that were installed for a previous version of the .NET SDK.  If there are workloads that were previously installed, we should also print a message as part of the first run experience telling the developer how to update them.
+
+We will use the `dotnet workload update` command for this.  This means that if there are any workload installation records for the current SDK feature band, it will update them.  Otherwise, it will use the workloads installation records from the most recent SDK feature band prior to the current one that has any workload installation records, and install those workloads.
+
+We will print a message on first run suggesting updating the workloads in the following cases:
+
+- A new feature band of the SDK was installed.  This will be if the following are both true:
+  - There are no workload installation records for the current SDK feature band
+  - There are workload installation records for any SDK feature band prior to the current one
+- A patch to the SDK was installed which updated the baseline workload manifest.  This is likely to happen for workloads that are shipped in sync with the .NET SDK, such as Blazor.  We will detect this by:
+  - From IAL: Get list of workload installation records for the current SDK
+  - Map these via the workload manifests to workload packs that should be installed
+  - Check if all those packs are installed.  If any aren't then print the the message
+
+## Updating the workload manifests
+
+The workload manifests describe available workloads and what workload packs make up each workload.
+
+A baseline version of each manifest will be included in the .NET SDK, which will allow the SDK to notify the user if a workload is needed to build a project.  However, these baseline manifest versions will not necessarily have the latest workload pack versions.  This is so that workloads don't have to insert each build into the .NET SDK, effectively linking the build graphs of all the workloads with the .NET SDK build graph.
+
+Because the manifests may be out of date (especially the baseline manifest), the workload manifests will by default be updated to the latest version when installing or updating workloads.
```

Comment:
How is this going to work for workload versions which are incompatible with the current SDK?

Reply:
This will be the latest workload manifest for the current SDK feature band.  So a 6.0.100 SDK wouldn't get a workload manifest update for 6.0.200.

Reply:
Does that mean that every manifest has to be versioned the same way as SDK?

Reply:
Workload manifests can patch separately from the SDK.  They do version along with the SDK feature band.